### ISSUE TITLE
Empty artists list and home page

### ISSUE THREAD

**Issue by mheppner** [airsonic/airsonic#681]
## Problem description

After a fresh installation of AirSonic and waiting for a single media folder to scan, both the sidebar and main home page are empty, despite artists being loaded in. I've tried reinstalling multiple times, using different database connectors.

[Possibly related to this issue](https://www.reddit.com/r/airsonic/comments/7qlpyp/issue_with_installation_airsonic_homeview_is_empty/), also on FreeBSD

### Steps to reproduce

I'm honestly at a loss for what the issue is or how to reproduce it.

- Nothing in the browser console, even with debug levels turned on.
- Nothing obvious in the Java logs, other than a bunch of `org.apache.catalina.connector.ClientAbortException: java.io.IOException: Broken pipe` exceptions, which seem like a client error to me.
- Nothing in the Java logs pop up during the media scan.
- Tried using the default database, as well as Postgres, both with the same result.
- Nginx config is direct from the docs.
- Media folder is mounted read-only, which was the same setup I had for Subsonic.
- Nothing is ever returned from search.
- Doing a manual SQL count in Postgres results in the same number as displayed in the sidebar below.
- Songs, artists, and albums are counted in the left sidebar, but nothing ever is displayed:

![screen shot 2018-01-19 at 9 48 04 pm](https://user-images.githubusercontent.com/2390428/35179177-8bc42d70-fd62-11e7-887d-a71aee4ae95b.png)

## System information

 * **Airsonic version**: *10.1.1-RELEASE – December 16, 2017*
 * **Operating system**: *Apache Tomcat/8.5.23, java 1.8.0_152, FreeBSD (237.0 MB / 316.0 MB)*
 * **Proxy server**: *Nginx*
 * **Client**: *Firefox 57.0.4*
 * **Postgres version**: *PostgreSQL 9.6.6 on amd64-portbld-freebsd11.1, compiled by FreeBSD clang version 4.0.0 (tags/RELEASE_400/final 297347) (based on LLVM 4.0.0), 64-bit*
 * **JDBC driver**: *org.postgresql.Driver*
 * **JDBC URL**: *jdbc:postgresql://<host>:5432/<dbname>?stringtype=unspecified*

**Comment by jooola** [comment]
We currently have some trouble to provide a proper proxy configuration, but could you tell me if the content is loaded or if its blocked by the "Mixed content policy" ?

Use the Firefox console to see if it blocks some requests (CTRL + SHIFT + E  and refresh the page)

**Comment by jooola** [comment]
And did you added this to your nginx configuration : 
`add_header Content-Security-Policy "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' www.gstatic.com; img-src 'self' *.akamaized.net; style-src 'self' 'unsafe-inline' fonts.googleapis.com; font-src 'self' fonts.gstatic.com; frame-src 'self'; object-src 'none'";`

**Comment by jooola** [comment]
@muff1nman Would you mind taking a quick look to this, I don't think this is a proxy related problem, and @KingDuckZ also had a similar issue when he tried to upgrade to latest 10.1.1.

**Comment by mheppner** [comment]
There is nothing in the FF or Chrome consoles, even with debug level enabled.

Here's the relevant part of my nginx config that I've been using with Subsonic for years: 

```
    location @subsonic {
        internal;
        proxy_pass         http://10.1.1.3:4040;
        proxy_next_upstream error timeout invalid_header http_500 http_502 http_503 http_404;
        proxy_buffering off;
        proxy_set_header Host $http_host;
        proxy_set_header X-Forwarded-Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto https;
        proxy_max_temp_file_size 0;
        proxy_redirect http:// https://;

        # added after upgrading to airsonic
        add_header       Strict-Transport-Security "max-age=31536000";
        add_header Content-Security-Policy "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' www.gstatic.com; img-src 'self' *.akamaized.net; style-src 'self' 'unsafe-inline' fonts.googleapis.com; font-src 'self' fonts.gstatic.com; frame-src 'self'; object-src 'none'";
    }

    location / {
        try_files $uri @subsonic;
    }
```

I don't think it's a proxy issue though, since the REST API is also returning 0 artists. The database definitely has been populated though.

**Comment by jooola** [comment]
@mheppner  Could you try with version 10.1.0 and 10.0.1 please ?

**Comment by mheppner** [comment]
Hmm. I turned on debug logging, and tried both versions, still the same result.

However, I went back to 10.1.1 and I tried disabling the folder, hitting save, then deleting the folder, save, re-adding it, and save. Then after starting a scan, things finally appeared in the sidebar. With debug level on, the logs rotate out too quickly to be able to find anything useful. It's only saving 8 log files...can I prevent that?

Unfortunately this is about all that happens. The artists appear in the sidebar, but when clicking on an artist, nothing is displayed except for the LastFM info. I opened an artist page in a new tab (below) and saved only the [logs](https://github.com/airsonic/airsonic/files/1662197/alabama_shakes_page.log) from when this request starts. It just looks like a bunch of SQL queries and URL routing to me. The folder containing this artist contains 3 subfolders with just mp3, flac, and jpg files, totaling ~550 MB. This folder also existed from when I used original Subsonic.

![screen shot 2018-01-24 at 7 58 40 pm](https://user-images.githubusercontent.com/2390428/35365203-256e5e5a-0141-11e8-9dd4-bdb852b43131.png)

**Comment by zorun** [comment]
I have the same issue with a fresh 10.1.1 install with OpenJDK 8 on Debian stretch.

Initial music import took a while to complete (I have around 40k songs).  But even after it's finished for several hours, there is absolutely no music, either in the web interface or through an Android app.  The only indication that something happened are the stats on the left which look correct:

![2018-01-25-100850_1359x660_scrot](https://user-images.githubusercontent.com/731144/35379852-cdc76fa8-01b7-11e8-884a-23053bdf862f.png)

**Comment by zorun** [comment]
Downgrading to 10.0.1, deleting the music folder and adding it again, and it works! (I haven't tried 10.1.0)

With 10.1.1:
```
Jan 25 10:30:52 subsonic java[869]: 2018-01-25 10:30:52.219  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 41750 entries.
Jan 25 10:30:52 subsonic java[869]: 2018-01-25 10:30:52.593  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 42000 entries.
Jan 25 10:30:53 subsonic java[869]: 2018-01-25 10:30:53.025  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 42250 entries.
Jan 25 10:30:53 subsonic java[869]: 2018-01-25 10:30:53.483  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 42500 entries.
Jan 25 10:30:53 subsonic java[869]: 2018-01-25 10:30:53.722  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 42654 entries.
Jan 25 10:30:53 subsonic java[869]: 2018-01-25 10:30:53.722  INFO --- o.a.p.s.MediaScannerService              : Marking non-present files.
Jan 25 10:30:53 subsonic java[869]: 2018-01-25 10:30:53.784  INFO --- o.a.p.s.MediaScannerService              : Marking non-present artists.
Jan 25 10:30:53 subsonic java[869]: 2018-01-25 10:30:53.791  INFO --- o.a.p.s.MediaScannerService              : Marking non-present albums.
Jan 25 10:30:53 subsonic java[869]: 2018-01-25 10:30:53.879  INFO --- o.a.p.s.MediaScannerService              : Completed media library scan.
Jan 25 10:30:54 subsonic java[869]: 2018-01-25 10:30:54.645  INFO --- o.a.p.service.PlaylistService            : Starting playlist import.
Jan 25 10:30:54 subsonic java[869]: 2018-01-25 10:30:54.646  INFO --- o.a.p.service.PlaylistService            : Completed playlist import.
```

With 10.0.1:
```
Jan 25 10:33:14 subsonic java[869]: 2018-01-25 10:33:14.503  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 42000 entries.
Jan 25 10:33:14 subsonic java[869]: 2018-01-25 10:33:14.790  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 42250 entries.
Jan 25 10:33:15 subsonic java[869]: 2018-01-25 10:33:15.056  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 42500 entries.
Jan 25 10:33:15 subsonic java[869]: 2018-01-25 10:33:15.227  INFO --- o.a.p.s.MediaScannerService              : Scanned media library with 42654 entries.
Jan 25 10:33:15 subsonic java[869]: 2018-01-25 10:33:15.227  INFO --- o.a.p.s.MediaScannerService              : Marking non-present files.
Jan 25 10:33:15 subsonic java[869]: 2018-01-25 10:33:15.261  INFO --- o.a.p.s.MediaScannerService              : Marking non-present artists.
Jan 25 10:33:15 subsonic java[869]: 2018-01-25 10:33:15.262  INFO --- o.a.p.s.MediaScannerService              : Marking non-present albums.
Jan 25 10:33:15 subsonic java[869]: 2018-01-25 10:33:15.296  INFO --- o.a.p.s.MediaScannerService              : Completed media library scan.
Jan 25 10:33:15 subsonic java[869]: 2018-01-25 10:33:15.704  INFO --- o.a.p.service.PlaylistService            : Starting playlist import.
Jan 25 10:33:15 subsonic java[869]: 2018-01-25 10:33:15.705  INFO --- o.a.p.service.PlaylistService            : Completed playlist import.
Jan 25 10:33:26 subsonic java[869]: 2018-01-25 10:33:26.302  INFO --- o.a.p.c.CoverArtController               : Created thumbnail cache /var/airsonic/thumbs/160
```

Not much difference, 10.1.1 is simply missing the last line about the thumbnail cache.

**Comment by gryphonmyers** [comment]
I experienced something similar setting up my server and it turned out I just needed to give my user access to the media folder.

**Comment by manglaneso** [comment]
I'm affected with this too. Downgrading to 10.0.1 seems to work.

**Comment by jooola** [comment]
@gryphonmyers So allowing your user to access the music folders fixed this for you ?

What about others ? Is this same issue ? @manglaneso ? @zorun @mheppner 

Could I get a proper log file with a higher logging level ?

**Comment by manglaneso** [comment]
I don't think is the same issue, if it were, then Airsonic would not be able to access the files, then the left sidebar would not display any information, right? Also, I just downgrade to 10.0.1 without any changes on the filesystem and it seems to work so far just by deleting the music folder and adding it again as @zorun suggested.

I can try to reproduce the error on another machine to get some logs if you want.

**Comment by jooola** [comment]
Yup logs would be great !

**Comment by muggenhor** [comment]
I'm experiencing the same problem and had reported it as #707 (I couldn't find a duplicate through search).

Some additional notes though:
 * I _can_ access music through DLNA (didn't verify that everything is accessible though) with "VLC for Android"
 * I also have my music directory mounted read-only (actually `mount -o bind,to src target`

**Comment by muggenhor** [comment]
I don't know how much help these reproduction steps are, but this is all I've got. To arrive at this problem I did:

1. follow the stand-alone tutorial (with Debian pkgs and systemd)
2. start airsonic (`systemctl start airsonic`)
3. change the admin password to something non-default
4. go to media folders and discover that it already has `/var/music` listed there, I had mine mounted on `/data/music` but I liked the `/var/music` path better so I shutdown airsonic, moved the mountpoint to that location and restart
5. I did see, in the webinterface, progress information about the scan
6. neither during the scan nor after it was finished did anything appear in the web interface

**Comment by jooola** [comment]
... So this is getting ridiculous...
You need to check 2 things : 


**The media folder is enabled:**
![image](https://user-images.githubusercontent.com/19195485/36949551-43e79140-1fea-11e8-8c7b-5e4c29ab63cf.png)

**The media folder is available for the user:**
![image](https://user-images.githubusercontent.com/19195485/36949555-513104e4-1fea-11e8-9b33-90a347925a45.png)

I managed to get the same screen as @mheppner posted in the first place : 
![image](https://user-images.githubusercontent.com/19195485/36949539-2614d3a8-1fea-11e8-98cc-4597ff645088.png)

So some here possibly have this as mentionned by @gryphonmyers https://github.com/airsonic/airsonic/issues/681#issuecomment-361076352

Still waiting some feedback If someone have the right media folders set but still have no access to his music.

**Comment by muggenhor** [comment]
@jooola I would not expect to see totals (artists, albums, etc) include folders you don't have access to

Also I had been trying with the admin user (in the web interface). Doesn't that user, by default, have access to the music directory that gets created by default?

**Comment by jooola** [comment]
@muggenhor Yup about left frame details this will be for another ticket.

No you can disable folder for admin so you  need to be sure it has access.

**Comment by muggenhor** [comment]
After having deleted and recreated `/var/music` I'm no longer able to reproduce this. Even when deleting the config and starting scratch. It may be related to the directory not having existed the first time I started airsonic while it did the second time. Except it's at least more complicated than that, because _just_ that doesn't work as a reproduction scenario.

At least the problem doesn't bite me anymore...

**Comment by RomanHargrave** [comment]
Having this issue with an imported config from subsonic. Same as pictured in the original description - stats show a ton of albums but page lists none.

**Comment by matoro** [comment]
Just ran into this.  The issue is when you create a non-admin user *before* adding music locations, which requires you to allow access to the music directory to the new user.  Extremely non-intuitive and should either be clearly documented or prompted (e.g. you just created a new user, would you like to give them access to all your music sources?)

**Comment by peterdstone** [comment]
I had the same issue on a new install.

[https://github.com/airsonic/airsonic/issues/681#issuecomment-370256398](url) Fixed it

The admin user isn't given access to the music libraries as they are added so this is a required step.

**Comment by eharris** [comment]
PR #1030 attempts to fix this by giving all Admin users access to all folders.

**Comment by sic6SaNdMaN** [comment]
I found this becaus of the following point:
"Media folder is mounted read-only, which was the same setup I had for Subsonic."
I can confirm, that airsonic needs media folders to be writable - for some reason.

I mounted a USB drive (FAT32) from my car with "mount -oro", because why should it be writable?
After many hours of trying around with ozher mount options like "iocharset" and "codepage" it was the read-only option, that prevented airsonic from scanning this media source.
A quick mount without "-oro" (but still ther other mount options) and the scanner is still scanning the folder like a champ...

Please check this, because it should absolutely NOT be neccessary that the media folders a writable for airsonic to work.
Thanks.

**Comment by tesshucom** [comment]
It seems that it has been reflected in master. Please close it when solved.

**Comment by sic6SaNdMaN** [comment]
@tesshucom Could you please provide a link to the issue or rhe commit, or whatever it is, that reflects this issue here?
Thanks.

**Comment by tesshucom** [comment]
> @tesshucomこの問題へのリンク、またはコミット、またはこの問題を反映しているコミットメントをここに提供していただけませんか？
> ありがとう。

Sorry. It seems that there was a mistake.
Change the label.

**Comment by antonpiatek** [comment]
> I experienced something similar setting up my server and it turned out I just needed to give my user access to the media folder. EDIT: I realize that description was extremely vague: within Airsonic, I needed to edit the user I had created for myself to "allow access" to the media folder I had set up.

I've just installed airsonic and hit the same issue, default admin user did not have access to the media folder - perhaps there is a default that should have been created somewhere in setup when adding the default folder location

**Comment by mitchellurgero** [comment]
For those that have this issue, the fix is easy:

go into Settings -> Users and `Allow Access to these Media Folders` and select the folder(s) that are not showing up for the user.

**Comment by makefu** [comment]
Hey, 1 year later and i have also encountered the same issue. Adding the share to admin user solved it. Maybe it would be helpful to add a short note in the first steps

**Comment by gimmelemons** [comment]
@makefu can you tell me how exactly you solved it? I'm having a similar issue, I gave my user permissions to /var/music directory, but still empty home.
I tried rescanning the folder, restarting docker.

**Comment by mitchellurgero** [comment]
> @makefu can you tell me how exactly you solved it? I'm having a similar issue, I gave my user permissions to /var/music directory, but still empty home.
> I tried rescanning the folder, restarting docker.

Go into settings then users and give admin access to the li early in there (it's a check box and drop down )

**Comment by gimmelemons** [comment]
![image](https://user-images.githubusercontent.com/65401076/109397625-9a257e80-795d-11eb-9adb-793cfc657595.png)
I did that too @mitchellurgero.

**Comment by gimmelemons** [comment]
This is how the docker volume permissions are when looking from the host machine:
```
drwxr-xr-x  3 root root  4096 Jan 20 08:46 .
drwx------ 20 root root  4096 Feb 27 20:17 ..
drwxr-xr-x  2 username username 77824 Feb 28 00:30 _data
```
This is the permissions when I exec into docker and ls:
```
rwxr-xr-x 1 root root   4096 Feb 27 18:16 .
drwxr-xr-x 1 root root   4096 Feb 27 18:16 ..
drwxr-xr-x 7 au   ag     4096 Feb 27 18:48 airsonic
drwxr-xr-x 2 root root   4096 Apr 15  2020 backups
drwxr-xr-x 1 root root   4096 Feb 26 21:17 cache
drwxr-xr-x 1 root root   4096 Feb 26 21:17 lib
drwxrwsr-x 2 root staff  4096 Apr 15  2020 local
lrwxrwxrwx 1 root root      9 Jan 19 01:01 lock -> /run/lock
drwxr-xr-x 1 root root   4096 Feb 26 21:17 log
drwxrwsr-x 2 root mail   4096 Jan 19 01:01 mail
drwxr-xr-x 2 au   ag    77824 Feb 27 19:00 music
drwxr-xr-x 2 root root   4096 Jan 19 01:01 opt
drwxr-xr-x 2 root root   4096 Jan 20 03:16 playlists
drwxr-xr-x 2 root root   4096 Jan 20 03:16 podcasts
lrwxrwxrwx 1 root root      4 Jan 19 01:01 run -> /run
drwxr-xr-x 2 root root   4096 Jan 19 01:01 spool
drwxrwxrwt 2 root root   4096 Jan 19 01:04 tmp
```
It feels like something is not right here.
